### ISSUE TITLE
SNS errors do not match what AWS actually returns

### ISSUE THREAD

**Issue by neoascetic** [spulec/moto#3477]
For example, when I am trying to fetch the attributes of not existing platform endpoint, moto gives me the following response:

```
<?xml version="1.0" encoding="UTF-8"?>
<ErrorResponse>
      <Errors>
        <Error>  
          <Code>NotFound</Code>
          <Message>Endpoint with arn arn:aws:sns:us-east-1:000000000000:endpoint/Bar/Foo/8c20d50c-82d5-4d39-93c8-100cc64fa512 not found</Message>
        </Error>
     </Errors>
     <RequestID>7a62c49f-347e-4fc4-9331-6e8eEXAMPLE</RequestID>
</ErrorResponse>
```

while AWS returns something like this:

```
<ErrorResponse xmlns="http://sns.amazonaws.com/doc/2010-03-31/">
  <Error>
    <Type>Sender</Type>
    <Code>NotFound</Code>
    <Message>Endpoint does not exist</Message>
  </Error>
  <RequestId>ac6c2567-45f5-5295-a271-7d49406cbe7c</RequestId>
</ErrorResponse>
```

I guess exceptions defined in https://github.com/spulec/moto/blob/master/moto/sns/exceptions.py must use [this template](https://github.com/spulec/moto/blob/19fc76f4669d545b56cfd79c55377f46379d546d/moto/sns/responses.py#L1095-L1102) to render themselves instead?

**Comment by bblommers** [comment]
Thanks for raising this @neoascetic!

Looks like you're right - the exceptions should define a custom template ([like this](https://github.com/spulec/moto/blob/master/moto/sagemaker/exceptions.py#L11)). Marking it as a bug.

**Comment by aantillonl** [comment]
I'd like to chip in with this issue, please check the PR and let me know your comments

**Comment by bblommers** [comment]
This is now fixed as of moto >= 1.3.17.dev235
